### ISSUE TITLE
cannot close window, keyboard events misrouted on MacOS Big Sur

### ISSUE THREAD

**Issue by bsmith94** [r0x0r/pywebview#839]
### Specification
- pywebview version: 3.5
- operating system: MacOS 11.6.2
- web renderer: not sure... default?

### Description  

I have a very basic test of pywebview. The window displays and shows the expected site. However

1. The window doesn't focus when it is created.
2. The window does not close when clicking on the frame's close button.  
3. Keyboard strokes appear in the terminal window that launched the python process instead of in the HTML form.

### Practicalities

- YES I am willing to work on this issue myself.

- NO I am prepared to support this issue financially.

Here is the code:

```
#!/usr/bin/env python3

import webview

w = webview.create_window('hello', 'https://www.duckduckgo.com')
webview.start()
```

**Comment by profittlich** [comment]
My experience is similar (pywebview 3.5, macOS 11.6.0, python 3.10.0). The window does focus and I can enter a search query, but the close button does not work (the closing callback is called, however). Using the menu to quit or CMD-Q closes the window, but it kills the process before the closed callback is finished and the closing callback does not get called at all:
```
import webview
import time

def onclosing():
        print("Shown only when clicking X on the window title bar, but the window does not close.")

def onclosed():
        print("Shown")
        time.sleep(1.0)
        print("Not shown")

w = webview.create_window('hello', 'https://www.duckduckgo.com')
w.closing += onclosing
w.closed += onclosed

webview.start()
```

**Comment by profittlich** [comment]
Same behavior with the current master.

**Comment by profittlich** [comment]
Correction: With master, the window closes correctly with the X button, but going though CMD-Q still does not work as expected.

**Comment by profittlich** [comment]
I'll open a separate issue for the cmd-Q behavior, as this seems unrelated to bsmith94's issues.

**Comment by r0x0r** [comment]
@bsmith94 I could not reproduce your issue. Works as expected both with 3.5 and latest master.

**Comment by bsmith94** [comment]
> @bsmith94 I could not reproduce your issue. Works as expected both with 3.5 and latest master.

Still broken in my world. Just tested the code from master. :( Wish I had more information for you.

**Comment by bsmith94** [comment]
I wouldn't be surprised if it isn't some weird config setting in Big Sur causing this. Apple really screwed the pooch with this OS.

**Comment by awesomo4000** [comment]
Hi. This commit in my fork made Ctrl-C and Cmd-Q work better on OSX (not sure if it applies to the version mentioned):

https://github.com/awesomo4000/pywebview/commit/b33d4ad26b785a35796c81fc88a152d5a6812c41


Also, make sure you're using a Framework version of Python. I build python using pyenv, and use the following to compile it:

```
env SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk \
MACOSX_DEPLOYMENT_TARGET=10.15  \
PYTHON_CONFIGURE_OPTS="--enable-framework" \
pyenv install -v 3.10.0
```

**Comment by bsmith94** [comment]
@awesomo4000 no dice on installing pyweb with that version of python. Same results with MacOSX10.15.sdk and MacOSX12.1.sdk. xcode is 13.2.1.

Looks like the issue is in webkit... but I can't verify that pyweb works with that version of python.

```
$ env SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/ MACOSX_DEPLOYMENT_TARGET=12.1  PYTHON_CONFIGURE_OPTS="--enable-framework" pyenv install -v 3.10.0
# python 3.10.0 installed
$ cd pyweb
$ $ /Users/bsmith/.pyenv/versions/3.10.0/bin/python3 setup.py install
running install
running bdist_egg
running egg_info
writing pywebview.egg-info/PKG-INFO
writing dependency_links to pywebview.egg-info/dependency_links.txt
writing requirements to pywebview.egg-info/requires.txt
writing top-level names to pywebview.egg-info/top_level.txt
reading manifest file 'pywebview.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE.md'
writing manifest file 'pywebview.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-12.1-x86_64/egg
running install_lib
running build_py
creating build/bdist.macosx-12.1-x86_64/egg
creating build/bdist.macosx-12.1-x86_64/egg/webview
copying build/lib/webview/screen.py -> build/bdist.macosx-12.1-x86_64/egg/webview
copying build/lib/webview/event.py -> build/bdist.macosx-12.1-x86_64/egg/webview
creating build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/edgechromium.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/gtk.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/cef.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/qt.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/__init__.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/edgehtml.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/cocoa.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/winforms.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/platforms/mshtml.py -> build/bdist.macosx-12.1-x86_64/egg/webview/platforms
copying build/lib/webview/window.py -> build/bdist.macosx-12.1-x86_64/egg/webview
copying build/lib/webview/serving.py -> build/bdist.macosx-12.1-x86_64/egg/webview
copying build/lib/webview/util.py -> build/bdist.macosx-12.1-x86_64/egg/webview
creating build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/event.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/__init__.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/drag.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/api.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/css.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/npo.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/dom.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/js/alert.py -> build/bdist.macosx-12.1-x86_64/egg/webview/js
copying build/lib/webview/__init__.py -> build/bdist.macosx-12.1-x86_64/egg/webview
copying build/lib/webview/localization.py -> build/bdist.macosx-12.1-x86_64/egg/webview
creating build/bdist.macosx-12.1-x86_64/egg/webview/lib
copying build/lib/webview/lib/Microsoft.Toolkit.Forms.UI.Controls.WebView.dll -> build/bdist.macosx-12.1-x86_64/egg/webview/lib
copying build/lib/webview/lib/Microsoft.Web.WebView2.LICENSE.md -> build/bdist.macosx-12.1-x86_64/egg/webview/lib
creating build/bdist.macosx-12.1-x86_64/egg/webview/lib/x64
copying build/lib/webview/lib/x64/WebView2Loader.dll -> build/bdist.macosx-12.1-x86_64/egg/webview/lib/x64
copying build/lib/webview/lib/WebBrowserInterop.x64.dll -> build/bdist.macosx-12.1-x86_64/egg/webview/lib
copying build/lib/webview/lib/WebBrowserInterop.x86.dll -> build/bdist.macosx-12.1-x86_64/egg/webview/lib
creating build/bdist.macosx-12.1-x86_64/egg/webview/lib/x86
copying build/lib/webview/lib/x86/WebView2Loader.dll -> build/bdist.macosx-12.1-x86_64/egg/webview/lib/x86
copying build/lib/webview/lib/Microsoft.Web.WebView2.WinForms.dll -> build/bdist.macosx-12.1-x86_64/egg/webview/lib
copying build/lib/webview/lib/Microsoft.Toolkit.Forms.UI.Controls.WebView.LICENSE.md -> build/bdist.macosx-12.1-x86_64/egg/webview/lib
copying build/lib/webview/lib/Microsoft.Web.WebView2.Core.dll -> build/bdist.macosx-12.1-x86_64/egg/webview/lib
copying build/lib/webview/guilib.py -> build/bdist.macosx-12.1-x86_64/egg/webview
copying build/lib/webview/wsgi.py -> build/bdist.macosx-12.1-x86_64/egg/webview
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/screen.py to screen.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/event.py to event.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/edgechromium.py to edgechromium.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/gtk.py to gtk.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/cef.py to cef.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/qt.py to qt.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/__init__.py to __init__.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/edgehtml.py to edgehtml.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/cocoa.py to cocoa.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/winforms.py to winforms.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/platforms/mshtml.py to mshtml.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/window.py to window.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/serving.py to serving.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/util.py to util.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/event.py to event.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/__init__.py to __init__.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/drag.py to drag.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/api.py to api.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/css.py to css.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/npo.py to npo.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/dom.py to dom.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/js/alert.py to alert.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/__init__.py to __init__.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/localization.py to localization.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/guilib.py to guilib.cpython-310.pyc
byte-compiling build/bdist.macosx-12.1-x86_64/egg/webview/wsgi.py to wsgi.cpython-310.pyc
creating build/bdist.macosx-12.1-x86_64/egg/EGG-INFO
copying pywebview.egg-info/PKG-INFO -> build/bdist.macosx-12.1-x86_64/egg/EGG-INFO
copying pywebview.egg-info/SOURCES.txt -> build/bdist.macosx-12.1-x86_64/egg/EGG-INFO
copying pywebview.egg-info/dependency_links.txt -> build/bdist.macosx-12.1-x86_64/egg/EGG-INFO
copying pywebview.egg-info/requires.txt -> build/bdist.macosx-12.1-x86_64/egg/EGG-INFO
copying pywebview.egg-info/top_level.txt -> build/bdist.macosx-12.1-x86_64/egg/EGG-INFO
writing build/bdist.macosx-12.1-x86_64/egg/EGG-INFO/native_libs.txt
zip_safe flag not set; analyzing archive contents...
webview.__pycache__.util.cpython-310: module references __file__
webview.platforms.__pycache__.cef.cpython-310: module references __file__
webview.platforms.__pycache__.cocoa.cpython-310: module references __file__
creating 'dist/pywebview-3.5-py3.10.egg' and adding 'build/bdist.macosx-12.1-x86_64/egg' to it
removing 'build/bdist.macosx-12.1-x86_64/egg' (and everything under it)
Processing pywebview-3.5-py3.10.egg
removing '/Users/bsmith/.pyenv/versions/3.10.0/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pywebview-3.5-py3.10.egg' (and everything under it)
creating /Users/bsmith/.pyenv/versions/3.10.0/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pywebview-3.5-py3.10.egg
Extracting pywebview-3.5-py3.10.egg to /Users/bsmith/.pyenv/versions/3.10.0/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages
pywebview 3.5 is already the active version in easy-install.pth

Installed /Users/bsmith/.pyenv/versions/3.10.0/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pywebview-3.5-py3.10.egg
Processing dependencies for pywebview==3.5
Searching for pyobjc-framework-WebKit
Reading https://pypi.org/simple/pyobjc-framework-WebKit/
Downloading https://files.pythonhosted.org/packages/42/54/56762c7bcb878ef88a4cf1adad515768dd60bc3fa99083225f95be29c803/pyobjc-framework-WebKit-8.1.tar.gz#sha256=aea4876a6a71a4e032637212bbb6a08cc4913f4813dbc5a104b5175e12f562ba
Best match: pyobjc-framework-WebKit 8.1
Processing pyobjc-framework-WebKit-8.1.tar.gz
Writing /var/folders/t8/j9cy79ld48zg9s8pqz_6r6kr0000gp/T/easy_install-ijb952nk/pyobjc-framework-WebKit-8.1/setup.cfg
Running pyobjc-framework-WebKit-8.1/setup.py -q bdist_egg --dist-dir /var/folders/t8/j9cy79ld48zg9s8pqz_6r6kr0000gp/T/easy_install-ijb952nk/pyobjc-framework-WebKit-8.1/egg-dist-tmp-ypps_bry
warning: no directories found matching 'Doc'
warning: no directories found matching 'source-deps'
warning: no previously-included files matching '.DS_Store' found anywhere in distribution
warning: no previously-included files matching '*.pyc' found anywhere in distribution
warning: no previously-included files matching '*.pyo' found anywhere in distribution
warning: no previously-included files matching '*.so' found anywhere in distribution
no previously-included directories found matching 'Lib/*/*.dSYM'
no previously-included directories found matching 'PyObjCTest/*.dSYM'
In file included from Modules/_WebKit.m:3:
In file included from /Users/bsmith/.pyenv/versions/3.10.0/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10/Python.h:25:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:64:
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:93:16: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
        unsigned char   *_base;
                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:93:16: note: insert '_Nullable' if the pointer may be null
        unsigned char   *_base;
                        ^
                          _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:93:16: note: insert '_Nonnull' if the pointer should never be null
        unsigned char   *_base;
                        ^
                          _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:138:32: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
        int     (* _Nullable _read) (void *, char *, int);
                                          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:138:32: note: insert '_Nullable' if the pointer may be null
        int     (* _Nullable _read) (void *, char *, int);
                                          ^
                                           _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:138:32: note: insert '_Nonnull' if the pointer should never be null
        int     (* _Nullable _read) (void *, char *, int);
                                          ^
                                           _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:138:40: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
        int     (* _Nullable _read) (void *, char *, int);
                                                  ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:138:40: note: insert '_Nullable' if the pointer may be null
        int     (* _Nullable _read) (void *, char *, int);
                                                  ^
                                                   _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:138:40: note: insert '_Nonnull' if the pointer should never be null
        int     (* _Nullable _read) (void *, char *, int);
                                                  ^
                                                   _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:139:35: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
        fpos_t  (* _Nullable _seek) (void *, fpos_t, int);
                                          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:139:35: note: insert '_Nullable' if the pointer may be null
        fpos_t  (* _Nullable _seek) (void *, fpos_t, int);
                                          ^
                                           _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:139:35: note: insert '_Nonnull' if the pointer should never be null
        fpos_t  (* _Nullable _seek) (void *, fpos_t, int);
                                          ^
                                           _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:140:32: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
        int     (* _Nullable _write)(void *, const char *, int);
                                          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:140:32: note: insert '_Nullable' if the pointer may be null
        int     (* _Nullable _write)(void *, const char *, int);
                                          ^
                                           _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:140:32: note: insert '_Nonnull' if the pointer should never be null
        int     (* _Nullable _write)(void *, const char *, int);
                                          ^
                                           _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:140:46: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
        int     (* _Nullable _write)(void *, const char *, int);
                                                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:140:46: note: insert '_Nullable' if the pointer may be null
        int     (* _Nullable _write)(void *, const char *, int);
                                                        ^
                                                         _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:140:46: note: insert '_Nonnull' if the pointer should never be null
        int     (* _Nullable _write)(void *, const char *, int);
                                                        ^
                                                         _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:144:18: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
        struct __sFILEX *_extra; /* additions to FILE to not break ABI */
                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:144:18: note: insert '_Nullable' if the pointer may be null
        struct __sFILEX *_extra; /* additions to FILE to not break ABI */
                        ^
                          _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/_stdio.h:144:18: note: insert '_Nonnull' if the pointer should never be null
        struct __sFILEX *_extra; /* additions to FILE to not break ABI */
                        ^
                          _Nonnull
In file included from Modules/_WebKit.m:3:
In file included from /Users/bsmith/.pyenv/versions/3.10.0/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10/Python.h:25:
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:67:13: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
extern FILE *__stdinp;
            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:67:13: note: insert '_Nullable' if the pointer may be null
extern FILE *__stdinp;
            ^
              _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:67:13: note: insert '_Nonnull' if the pointer should never be null
extern FILE *__stdinp;
            ^
              _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:378:41: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
                 int (* _Nullable)(void *, const char *, int),
                                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:378:41: note: insert '_Nullable' if the pointer may be null
                 int (* _Nullable)(void *, const char *, int),
                                        ^
                                         _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:378:41: note: insert '_Nonnull' if the pointer should never be null
                 int (* _Nullable)(void *, const char *, int),
                                        ^
                                         _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:378:55: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
                 int (* _Nullable)(void *, const char *, int),
                                                      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:378:55: note: insert '_Nullable' if the pointer may be null
                 int (* _Nullable)(void *, const char *, int),
                                                      ^
                                                       _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:378:55: note: insert '_Nonnull' if the pointer should never be null
                 int (* _Nullable)(void *, const char *, int),
                                                      ^
                                                       _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:379:44: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
                 fpos_t (* _Nullable)(void *, fpos_t, int),
                                           ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:379:44: note: insert '_Nullable' if the pointer may be null
                 fpos_t (* _Nullable)(void *, fpos_t, int),
                                           ^
                                            _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:379:44: note: insert '_Nonnull' if the pointer should never be null
                 fpos_t (* _Nullable)(void *, fpos_t, int),
                                           ^
                                            _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:380:41: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
                 int (* _Nullable)(void *));
                                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:380:41: note: insert '_Nullable' if the pointer may be null
                 int (* _Nullable)(void *));
                                        ^
                                         _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:380:41: note: insert '_Nonnull' if the pointer should never be null
                 int (* _Nullable)(void *));
                                        ^
                                         _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:376:6: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
FILE    *funopen(const void *,
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:376:6: note: insert '_Nullable' if the pointer may be null
FILE    *funopen(const void *,
        ^
          _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdio.h:376:6: note: insert '_Nonnull' if the pointer should never be null
FILE    *funopen(const void *,
        ^
          _Nonnull
In file included from Modules/_WebKit.m:3:
In file included from /Users/bsmith/.pyenv/versions/3.10.0/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10/Python.h:34:
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:134:25: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
double   atof(const char *);
                         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:134:25: note: insert '_Nullable' if the pointer may be null
double   atof(const char *);
                         ^
                          _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:134:25: note: insert '_Nonnull' if the pointer should never be null
double   atof(const char *);
                         ^
                          _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:135:22: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
int      atoi(const char *);
                         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:135:22: note: insert '_Nullable' if the pointer may be null
int      atoi(const char *);
                         ^
                          _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:135:22: note: insert '_Nonnull' if the pointer should never be null
int      atoi(const char *);
                         ^
                          _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:136:23: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
long     atol(const char *);
                         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:136:23: note: insert '_Nullable' if the pointer may be null
long     atol(const char *);
                         ^
                          _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:136:23: note: insert '_Nonnull' if the pointer should never be null
long     atol(const char *);
                         ^
                          _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:139:20: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
         atoll(const char *);
                          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:139:20: note: insert '_Nullable' if the pointer may be null
         atoll(const char *);
                          ^
                           _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:139:20: note: insert '_Nonnull' if the pointer should never be null
         atoll(const char *);
                          ^
                           _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:141:26: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
void    *bsearch(const void *__key, const void *__base, size_t __nel,
                            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:141:26: note: insert '_Nullable' if the pointer may be null
void    *bsearch(const void *__key, const void *__base, size_t __nel,
                            ^
                              _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:141:26: note: insert '_Nonnull' if the pointer should never be null
void    *bsearch(const void *__key, const void *__base, size_t __nel,
                            ^
                              _Nonnull
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:141:45: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
void    *bsearch(const void *__key, const void *__base, size_t __nel,
                                               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:141:45: note: insert '_Nullable' if the pointer may be null
void    *bsearch(const void *__key, const void *__base, size_t __nel,
                                               ^
                                                 _Nullable
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk//usr/include/stdlib.h:141:45: note: insert '_Nonnull' if the pointer should never be null
void    *bsearch(const void *__key, const void *__base, size_t __nel,
                                               ^
                                                 _Nonnull
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
error: Setup script exited with error: command '/usr/bin/clang' failed with exit code 1
```

**Comment by awesomo4000** [comment]
I see that our SDK versions are different (I'm still using MacOSX10.15.sdk for deployment target 10.15 .) At any rate, I used the pyobjc `AppHelper.installMachInterrupt()` function to make Ctrl-c, cmd-q, and click red x work for me:

```
diff --git a/webview/platforms/cocoa.py b/webview/platforms/cocoa.py
index c5472236..43fd11b9 100755
--- a/webview/platforms/cocoa.py
+++ b/webview/platforms/cocoa.py
@@ -453,6 +453,7 @@ def first_show(self):
             self._add_view_menu()
 
             BrowserView.app.activateIgnoringOtherApps_(Foundation.YES)
+            AppHelper.installMachInterrupt()
             BrowserView.app.run()
 
     def show(self):
```

**Comment by r0x0r** [comment]
@awesomo4000 would you create a PR?

**Comment by r0x0r** [comment]
@awesomo4000's fix released in 3.6. I am closing this now.
